File: Challenges/Go/ex3.go
package main

import "fmt"
import "os"
import "crypto/md5"
import "bytes"

func main(){
        if len(os.Args) != 2{
                fmt.Println("Usage: ./prog input")
                fmt.Println("Hint: ./prog -h")
        }else if os.Args[1] == "-h"{
                fmt.Println("Hashes go great with eggs.")
        }else{
                sum1 := md5.Sum([]byte(os.Args[1]))
                bytes1 := sum1[0:len(sum1)]
                sum2 := [16]byte{27, 194, 155, 54, 246, 35, 186, 130, 170, 246, 114, 79, 211, 177, 103, 24}
                bytes2 := sum2[0:len(sum2)]
                if bytes.Compare(bytes1, bytes2) == 0{
                        fmt.Println("Score!")
                }else{
                        fmt.Println("Fail!")
                }
        }
}

File: Challenges/C/ex3.c
#include<stdio.h>
#include<string.h>
#include<stdlib.h>
int main(int argc, char** argv){
  if(argc!=2){
    printf("Usage: ./prog input\n");
    printf("Hint: ./prog -h\n");
  }else if(strcmp("-h",argv[1])==0){
    printf("Recall hex 0x30 is ascii digit 0, so 0x30+1 is ascii digit 1.\n");
  }else{
    char *psswd=malloc(sizeof(char)*10+1);
    for(int i=0;i<10;i++){
		psswd[i]=i+0x30;
    }
    psswd[10]='\0';
    strcmp(psswd,argv[1])==0?printf("Score!\n"):printf("Fail!\n");
  }
  return 0;
}
